src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -223,6 +223,25 @@ private ConstantValue DefaultSyntaxValue
         // in which case the declaring compilation is the wrong one.
         protected ConstantValue MakeDefaultExpression(DiagnosticBag diagnostics, Binder binder)
         {
+            if (this.IsNullChecked)
```

Comment:
>if (this.IsNullChecked) [](start = 12, length = 23)

It seems surprising we're reporting these diagnostics in `SourceComplexParameterSymbol.MakeDefaultExpression()`. Are all null-checked parameters represented with `SourceComplexParameterSymbol`, and does `MakeDefaultExpression` get called in all cases?

Reply:
It looks like `LambdaSymbol` does reports some of the same diagnostics (but does not report missing `System_ArgumentNullException__ctorString`).

The error reporting should be shared across the two places, and we should have a missing member test for a null-checked parameter in a lambda.

---
In reply to: [302761272](https://github.com/dotnet/roslyn/pull/36979#discussion_r302761272) [](ancestors = 302761272)

Reply:
I believe Simple and synthesized symbols are only produced in the lambda and local function cases. I believe I still have to add in the local functions ones. 

The lazy resolving of the parameters' nullability prevented me from doing it in the ParameterHelpers, so @agocke suggested using `MakeDefaultExpression` instead. 

I'll move over the missing member diagnostic, and add a test.

Reply:
Sounds reasonable. Ideally the two locations should use the same helper method, which probably belongs in ParameterHelpers.

---
In reply to: [302763413](https://github.com/dotnet/roslyn/pull/36979#discussion_r302763413) [](ancestors = 302763413)